gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb

```diff
@@ -5,89 +5,68 @@
+    #   * `~/.aws/config`
+    #
+    # @option options [Integer] :retries (3) Number of times to retry
     #   when retrieving credentials.
     # @option options [String] :endpoint ('http://169.254.169.254') The IMDS
-    #   endpoint. This option has precedence over the :endpoint_mode.
+    #   endpoint. This option has precedence over the `:endpoint_mode`.
     # @option options [String] :endpoint_mode ('IPv4') The endpoint mode for
     #   the instance metadata service. This is either 'IPv4' ('169.254.169.254')
     #   or 'IPv6' ('[fd00:ec2::254]').
-    # @option options [Boolean] :disable_imds_v1 (false) Disable the use of the
-    #  legacy EC2 Metadata Service v1.
+    # @option options [Boolean] :disable_imds_v1 (false) Deprecated. The legacy
+    #   EC2 Metadata Service v1 has been retired. Only IMDSv2 is supported.
     # @option options [String] :ip_address ('169.254.169.254') Deprecated. Use
-    #   :endpoint instead. The IP address for the endpoint.
+    #   `:endpoint` instead. The IP address for the endpoint.
     # @option options [Integer] :port (80)
     # @option options [Float] :http_open_timeout (1)
     # @option options [Float] :http_read_timeout (1)
-    # @option options [Numeric, Proc] :delay By default, failures are retried
+    # @option options [Numeric, Proc] :backoff By default, failures are retried
     #   with exponential back-off, i.e. `sleep(1.2 ** num_failures)`. You can
     #   pass a number of seconds to sleep between failed attempts, or
     #   a Proc that accepts the number of failures.
     # @option options [IO] :http_debug_output (nil) HTTP wire
     #   traces are sent to this object.  You can specify something
-    #   like $stdout.
+    #   like `$stdout`.
     # @option options [Integer] :token_ttl Time-to-Live in seconds for EC2
     #   Metadata Token used for fetching Metadata Profile Credentials, defaults
     #   to 21600 seconds
     # @option options [Callable] before_refresh Proc called before
     #   credentials are refreshed. `before_refresh` is called
-    #   with an instance of this object when
-    #   AWS credentials are required and need to be refreshed.
+    #   with an instance of this object when AWS credentials are required
+    #   and need to be refreshed.
     def initialize(options = {})
-      @retries = options[:retries] || 1
-      endpoint_mode = resolve_endpoint_mode(options)
-      @endpoint = resolve_endpoint(options, endpoint_mode)
-      @port = options[:port] || 80
-      @disable_imds_v1 = resolve_disable_v1(options)
-      # Flag for if v2 flow fails, skip future attempts
-      @imds_v1_fallback = false
-      @http_open_timeout = options[:http_open_timeout] || 1
-      @http_read_timeout = options[:http_read_timeout] || 1
-      @http_debug_output = options[:http_debug_output]
-      @backoff = backoff(options[:backoff])
-      @token_ttl = options[:token_ttl] || 21_600
-      @token = nil
+      @client = options[:client] || create_client(options)
```

Comment:
I think you want to do options.delete(:client) such that client doesn't get passed to ec2 metadata. You'll want to consider that for other options.